public/app/features/auth-config/types.ts

```diff
@@ -10,18 +11,63 @@ export interface AuthProviderInfo {
 
 export type GetStatusHook = () => Promise<AuthProviderStatus>;
 
+// Settings types common to the provider settings data when working with the API and forms
+export type SSOProviderSettingsBase = {
+  allowAssignGrafanaAdmin?: boolean;
+  allowSignup?: boolean;
```

Comment:
This field should be `allowSignUp` instead of `allowSignup`. This comes from the conversion we make on the backend from snake_case to camelCase. On the backend its name is `allow_sign_up` and the conversion will rename it to `allowSignUp`.